local Types = {}

export type Canvas = {
	OutputWarnings: boolean,
	AutoRender: boolean,
	AutoRenderFpsLimit: number,
	AlphaBlendingMode: number,
	GridTransparency: number,
	GridColour: Color3,
	CanvasColour: Color3,
	ClearAlpha: number,
	Resolution: Vector2,
	CurrentResX: number,
	CurrentResY: number,
	Width: number,
	Height: number,
	FrameSize: Vector2,
	InternalResolution: Vector2,
	Parent: GuiObject,
	Frame: Frame,
	Container: Frame,
	ImageLabel: ImageLabel,
	EditableImage: EditableImage?,
	Buffer: buffer,
	NeedsRender: boolean,
	EventsFolder: Folder,
	OnRendered: RBXScriptSignal,
	AutoUpdateConnection: RBXScriptConnection,

	SetBuffer: (Canvas, buffer, number?, number?, number?, number?) -> (),
	GetBuffer: (Canvas, number?, number?, number?, number?) -> buffer,
	SetRGB: (Canvas, number, number, number, number, number) -> (),
	SetRGBA: (Canvas, number, number, number, number, number, number?) -> (),
	SetPixel: (Canvas, any, any, Color3?, number?) -> (),
	SetAlpha: (Canvas, number, number, number) -> (),
	SetU32: (Canvas, number, number, number) -> (),
	GetRGB: (Canvas, number, number) -> (number, number, number),
	GetRGBA: (Canvas, number, number) -> (number, number, number, number),
	GetAlpha: (Canvas, number, number) -> number,
	GetU32: (Canvas, number, number) -> number,
	GetPixel: (Canvas, Vector2) -> Color3,
	GetPixelXY: (Canvas, number, number) -> Color3,
	FillRGBA: (Canvas, number, number, number, number?) -> (),
	Fill: (Canvas, Color3, number?) -> (),
	Clear: (Canvas) -> (),
	SetClearRGBA: (Canvas, number, number, number, number) -> (),
	DrawLine: (Canvas, Vector2, Vector2, Color3, number?, boolean?) -> (),
	DrawRectangle: (Canvas, Vector2, Vector2, Color3, number?, boolean?) -> (),
	DrawRect: (Canvas, number, number, number, number, Color3, number?) -> (),
	Resize: (Canvas, any, number?) -> (),
	Render: (Canvas) -> (),
	SetBlur: (Canvas, boolean) -> (),
	SetStretchToFit: (Canvas, boolean) -> (),
	ViewportToCanvasPoint: (Canvas, Vector2) -> Vector2,
	GetMousePoint: (Canvas) -> Vector2?,
	MouseIsOnTop: (Canvas) -> boolean,
	Destroy: (Canvas) -> (),
}

export type Options = {
	Parent: GuiObject,
	Resolution: Vector2?,
	FrameSize: Vector2?,
	FrameWidth: number?,
	FrameHeight: number?,
	Width: number?,
	Height: number?,
	CanvasColour: Color3?,
	BackgroundColor: Color3?,
	Blur: boolean?,
	Name: string?,
}

return Types
